plugins/inputs/ctrlx_datalayer/ctrlx_datalayer_test.go

```diff
@@ -0,0 +1,256 @@
+	"github.com/influxdata/telegraf/config"
+	httpconfig "github.com/influxdata/telegraf/plugins/common/http"
+	"github.com/influxdata/telegraf/plugins/common/tls"
+	"github.com/influxdata/telegraf/testutil"
+	"github.com/stretchr/testify/require"
+)
+
+const path = "/automation/api/v2/events"
+
+var multiEntries = false
+var mux sync.Mutex
+
+func setMultiEntries(m bool) {
+	mux.Lock()
+	defer mux.Unlock()
+	multiEntries = m
+}
+
+func getMultiEntries() bool {
+	mux.Lock()
+	defer mux.Unlock()
+	return multiEntries
+}
+
+func TestCtrlXCreateSubscriptionBasic(t *testing.T) {
+	server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		w.WriteHeader(http.StatusCreated)
+		_, err := w.Write([]byte("201 created"))
+		require.NoError(t, err)
+	}))
+	defer server.Close()
+
+	subs := make([]Subscription, 0)
+	subs = append(subs, Subscription{
+		index: 0,
+		Nodes: []Node{
+			{Name: "counter", Address: "plc/app/Application/sym/PLC_PRG/counter"},
+			{Name: "counterReverse", Address: "plc/app/Application/sym/PLC_PRG/counterReverse"},
+		},
+	},
+	)
+	s := &CtrlXDataLayer{
+		connection:   &http.Client{},
+		url:          server.URL,
+		Username:     config.NewSecret([]byte("user")),
+		Password:     config.NewSecret([]byte("password")),
+		tokenManager: token.TokenManager{Url: server.URL, Username: "user", Password: "password", Connection: &http.Client{}},
+		Subscription: subs,
+		Log:          testutil.Logger{},
+	}
+
+	subID, err := s.createSubscription(&subs[0])
+
+	require.NoError(t, err)
+	require.NotEmpty(t, subID)
+}
+
+func TestCtrlXCreateSubscriptionDriven(t *testing.T) {
+	var tests = []struct {
+		res    string
+		status int
+		subID  string
+	}{
+		{res: "{\"status\":200}", status: 200, subID: "subID"},
+		{res: "{\"status\":401}", status: 401, subID: ""},
+	}
+
+	for _, test := range tests {
+		t.Run(test.res, func(t *testing.T) {
+			server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+				w.WriteHeader(test.status)
+				_, err := w.Write([]byte(test.res))
+				require.NoError(t, err)
+			}))
+			defer server.Close()
+			subs := make([]Subscription, 0)
+			subs = append(subs, Subscription{
+				Nodes: []Node{
+					{Name: "counter", Address: "plc/app/Application/sym/PLC_PRG/counter"},
+					{Name: "counterReverse", Address: "plc/app/Application/sym/PLC_PRG/counterReverse"},
+				},
+			},
+			)
+			s := &CtrlXDataLayer{
+				connection:   &http.Client{},
+				url:          server.URL,
+				Username:     config.NewSecret([]byte("user")),
+				Password:     config.NewSecret([]byte("password")),
+				Subscription: subs,
+				tokenManager: token.TokenManager{Url: server.URL, Username: "user", Password: "password", Connection: &http.Client{}},
+				Log:          testutil.Logger{},
+			}
+			subID, err := s.createSubscription(&subs[0])
+
+			if test.subID != "" {
+				require.NoError(t, err)
+				require.NotEmpty(t, subID)
```

Comment:
Should the received subID be the same as the one in the test? Maybe do a `require.Equal` then.